electrode_models/metal_air_discret.py

```diff
@@ -0,0 +1,476 @@
+            - SVdot: the time derivative of each state variable: dSV/dt
+            - electrode: the object representing the current electrode
+            - sep: the object representing the separator
+            - counter: the object representing the electrode counter to the current electrode
+            - params: dict of battery simulation parameters.
+        """
+        # Initialize the residual:
+        resid = np.zeros((self.n_vars_tot,))
+
+        # Save local copies of the solution vectors, pointers for this electrode:
+        SVptr = self.SVptr
+        SV_loc = SV[SVptr['electrode']]
+        SVdot_loc = SVdot[SVptr['electrode']]
+       
+        j = 0
+        # Read out properties:
+        phi_ed = SV_loc[SVptr['phi_ed'][j]]
+        print("phi_ed", phi_ed)
+        phi_elyte = phi_ed + SV_loc[SVptr['phi_dl'][j]]
+        print("phi_elyte", phi_elyte)
+        # print('phi_elyte = ', phi_elyte)
+        c_k_elyte = SV_loc[SVptr['C_k_elyte'][j]]
+        eps_product = SV_loc[SVptr['eps_product'][j]]
+         # Set Cantera object properties:
+        self.host_obj.electric_potential = phi_ed
+        self.elyte_obj.electric_potential = phi_elyte
+        self.elyte_obj.X = c_k_elyte
+        # Set microstructure multiplier for effective diffusivities
+        eps_elyte = 1. - eps_product - self.eps_host
+        self.elyte_microstructure = eps_elyte**1.5
+        # Read electrolyte fluxes at the separator boundary:
+        N_k_sep, i_io_sep = sep.electrode_boundary_flux(SV, self, params['T'])
+
+        #calculate flux out    
+        phi_ed_next = SV_loc[SVptr['phi_ed'][j+1]]
+        phi_elyte_next = phi_ed_next + SV_loc[SVptr['phi_dl'][j+1]]
+        # print('phi_elyte = ', phi_elyte)
+        c_k_elyte_next = SV_loc[SVptr['C_k_elyte'][j+1]]
+        eps_product_next = SV_loc[SVptr['eps_product'][j+1]]
+        eps_elyte_next = 1. - eps_product_next- self.eps_host
+        
+        state_1 = {'C_k': c_k_elyte, 'phi':phi_elyte, 'T': params['T'], 'dy':self.dy, 
+            'microstructure':eps_elyte**1.5}
+        state_2 = {'C_k': c_k_elyte_next, 'phi':phi_elyte_next, 'T': params['T'], 'dy':self.dy, 
+            'microstructure':eps_elyte_next**1.5}
+
+        # Multiply by ed.i_ext_flag: fluxes are out of the anode, into the cathode.
+        N_k, i_io = sep.elyte_transport(state_1, state_2,sep)
+        i_el = (self.i_ext_flag * self.sigma_el*(phi_ed - phi_ed_next)*self.dyInv)
+
+        if self.name=='anode':
+            # The electric potential of the anode = 0 V.
+            resid[[SVptr['phi_ed'][j]]] = phi_ed
+
+        elif self.name=='cathode':
+            # For the cathode, the potential of the cathode must be such that 
+            # the electrolyte electric potential (calculated as phi_ca + 
+            # dphi_dl) produces the correct ionic current between the separator # and cathode:
+            if params['boundary'] == 'current':
+                resid[SVptr['phi_ed'][j]] = i_io_sep - i_io - i_el
```

Comment:
I think let me merge this PR before you make this change, but this line (227) is the only thing you need, for this node.  This equation is the correct one, no matter if you are anode or cathode, potentiostatic or galvanostatic boundary condition. You can delete everything else associated with the `if/elif` decisions, here.

But like I said, don't change it yet. ;)